pkg/kubelet/prober/prober_manager.go

```diff
@@ -236,19 +236,33 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 			started = !exists
 		}
 		podStatus.ContainerStatuses[i].Started = &started
-
 		if started {
 			var ready bool
 			if c.State.Running == nil {
 				ready = false
-			} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok && result == results.Success {
-				ready = true
+			} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok {
+				if result == results.Success {
+					ready = true
+				} else {
+					ready = false
+				}
 			} else {
 				// The check whether there is a probe which hasn't run yet.
 				w, exists := m.getWorker(podUID, c.Name, readiness)
 				ready = !exists // no readinessProbe -> always ready
 				if exists {
+					// The readiness worker has not yet reported to readinessManager, instead of just returning ready=false consult status_Manager
```

Comment:
**What if it hasn't?**

Imagine a one replica persistent database behind a service having an endpoint, which will get deleted and added again only when you update the kubelet on the node -  a cascade of load on the API-Server and downtime for consumers of that service.

Nodes, which host one replica persistent databases, should run as long as possible hosting the workload since you cannot easily move around the workload without producing downtime. To make them run as long as possible, you will exchange kubelet from time to time to be on a recent version. Just by default considering all pods not-ready on startup does not sound desirable to me.

Maybe it is worth discussing if this can be enabled optionally?

**What if status has changed?**
If status has changed kubelet will eventually report the actual-ready state. Actually, we only report back the state from API-Server as long as no worker has done it's probe job.

Reply:
@ehashman Unfortunately, you have not replied anymore. We are struck by the same.